app/controllers/api/v1/users_controller.rb

```diff
@@ -55,7 +55,7 @@ def create
           render_data data: current_user, serializer: CurrentUserSerializer, status: :created
         else
           # TODO: amir - Improve logging.
-          render_error errors: user.errors.to_a, status: :bad_request
+          render_error errors: 'UserAlreadyExists', status: :bad_request
```

Comment:
So what this will do is always throw this error regardless of why it fails. We need to check `user.errors` and see if a duplicate email is the reason they fail. If not, we can just return `render_error status: :bad_request`

Reply:
I addressed the change in my latest commit :+1: